### PR TITLE
Fix deadlocks in front/lib/api/assistant/conversation.ts

### PR DESCRIPTION
**Pull request by lasryaric**
## Description

This PR removes all queries outside of transaction blocks that have issued a `pg_advisory_xact_lock`, which were the cause of deadlock (transaction waiting on a query, which is waiting on the connection pool to free up a spot, which is waiting on the transaction to complete).

### CODE REVIEW

File: front/lib/api/assistant/conversation.ts [hunk 1]
```diff
@@ -606,9 +613,19 @@ export async function* postUserMessage(
     return;
   }
 
+  const agentConfigurations = removeNulls(
+    await Promise.all(
+      mentions.filter(isAgentMention).map((mention) => {
+        return getAgentConfiguration(auth, mention.configurationId);
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/lib/api/assistant/conversation.ts:619]: Worth case we have an error below and there is an extra participant but I thiiiink that's fine?

**Comment by lasryaric** [comment]
@spolu agree but this is unrelated to the fix of this PR, and a message rarely have more than one or two agents mentions, so do we need to spend time on this?

**Comment by spolu** [comment]
I think the `createOrUpdateParticipation` move is directly related to this right?
For agentConfigurations, agreed it's not directly related but it's a good way to remove a Promise.all and build a better future no? So we might as well tackle it since it's a few lines of code?

**Comment by spolu** [comment]
In any case we def def def want a non full variant to save a bunch of DB requests on postUserMessage which is very aligned with the goal here?

**Comment by spolu** [comment]
I think we can all do it here as it's all small fixes. 

Otherwise, let's do the light version at the very least and create eng runner cards for the rest :+1:

**Comment by lasryaric** [comment]
I agree these are small changes but I also want to keep the PR clear. 100% of the problem comes from querying from within the transaction, so lets ship that and make a separate PR for the other optimizations.

**Review by spolu — APPROVED** [review]

**Comment by lasryaric** [comment]
Thanks for the approval, and sorry for persisting, but templating `getAgentConfiguration()` on top of the templated `getAgentConfigration[s]()` is actually pretty complicated :/ Looking into it for a follow up PR 👍

**Pull request merged**